### PR TITLE
refine spill failpoint and `SharedSpilledBucketDataLoader`

### LINKED ISSUE #7395: Pipeline: Tiflash crash when spill to disk
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2023/04/27 03:09:29.033 +08:00] [ERROR] [BaseDaemon.cpp:376] [########################################] [source=BaseDaemon] [thread_id=906]
[2023/04/27 03:09:29.035 +08:00] [ERROR] [BaseDaemon.cpp:377] ["(from thread 106) Received signal Aborted(6)."] [source=BaseDaemon] [thread_id=906]
[2023/04/27 03:09:29.072 +08:00] [ERROR] [BaseDaemon.cpp:569] ["\n       0xad204bd\tfaultSignalHandler(int, siginfo_t*, void*) [tiflash+181535933]\n                \tlibs/libdaemon/src/BaseDaemon.cpp:220\n  0x7f954c747d90\t<unknown symbol> [libc.so.6+347536]\n  0x7f954c79454c\t__pthread_kill_implementation [libc.so.6+660812]\n  0x7f954c747ce6\tgsignal [libc.so.6+347366]\n  0x7f954c71b7f3\t__GI_abort [libc.so.6+165875]\n  0x7f954c71b71b\t__assert_fail_base.cold [libc.so.6+165659]\n  0x7f954c740c66\t__assert_fail [libc.so.6+318566]\n       0xbfb5d73\tDB::SharedAggregateRestorer::tryLoadBucketData() [tiflash+201022835]\n                \tdbms/src/Operators/SharedAggregateRestorer.cpp:167\n       0xc0275c1\tDB::AggregateRestoreSourceOp::awaitImpl() [tiflash+201487809]\n                \tdbms/src/Operators/AggregateRestoreSourceOp.cpp:42\n       0xbfe28c9\tDB::Operator::await() [tiflash+201205961]\n                \tdbms/src/Operators/Operator.cpp:38\n       0xbfdeb3f\tDB::PipelineExec::awaitImpl() [tiflash+201190207]\n                \tdbms/src/Flash/Pipeline/Exec/PipelineExec.cpp:160\n       0xbfde8d5\tDB::PipelineExec::await() [tiflash+201189589]\n                \tdbms/src/Flash/Pipeline/Exec/PipelineExec.cpp:142\n       0xbfdbab0\tDB::PipelineTask::doAwaitImpl() [tiflash+201177776]\n                \tdbms/src/Flash/Pipeline/Schedule/Tasks/PipelineTask.cpp:101\n       0xbfd7aa9\tDB::EventTask::awaitImpl()::$_2::operator()() const [tiflash+201161385]\n                \tdbms/src/Flash/Pipeline/Schedule/Tasks/EventTask.cpp:79\n       0xbfd7a75\tdecltype(static_cast<DB::EventTask::awaitImpl()::$_2&>(fp)()) std::__1::__invoke<DB::EventTask::awaitImpl()::$_2&>(DB::EventTask::awaitImpl()::$_2&) [tiflash+201161333]\n                \t/usr/local/bin/../include/c++/v1/type_traits:3918\n       0xbfd7a2d\tDB::ExecTaskStatus std::__1::__invoke_void_return_wrapper<DB::ExecTaskStatus, false>::__call<DB::EventTask::awaitImpl()::$_2&>(DB::EventTask::awaitImpl()::$_2&) [tiflash+201161261]\n                \t/usr/local/bin/../include/c++/v1/__functional/invoke.h:30\n       0xbfd79fd\tstd::__1::__function::__alloc_func<DB::EventTask::awaitImpl()::$_2, std::__1::allocator<DB::EventTask::awaitImpl()::$_2>, DB::ExecTaskStatus ()>::operator()() [tiflash+201161213]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:171\n       0xbfd6c99\tstd::__1::__function::__func<DB::EventTask::awaitImpl()::$_2, std::__1::allocator<DB::EventTask::awaitImpl()::$_2>, DB::ExecTaskStatus ()>::operator()() [tiflash+201157785]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:345\n       0xbfd7ae6\tstd::__1::__function::__value_func<DB::ExecTaskStatus ()>::operator()() const [tiflash+201161446]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:498\n       0xbfd4175\tstd::__1::function<DB::ExecTaskStatus ()>::operator()() const [tiflash+201146741]\n                \t/usr/local/bin/../include/c++/v1/__functional/function.h:1175\n       0xbfd3d6b\tDB::EventTask::doTaskAction(std::__1::function<DB::ExecTaskStatus ()>&&) [tiflash+201145707]\n                \tdbms/src/Flash/Pipeline/Schedule/Tasks/EventTask.cpp:91\n       0xbfd40f0\tDB::EventTask::awaitImpl() [tiflash+201146608]\n                \tdbms/src/Flash/Pipeline/Schedule/Tasks/EventTask.cpp:79\n       0xbfe3ac9\tDB::Task::await() [tiflash+201210569]\n                \tdbms/src/Flash/Pipeline/Schedule/Tasks/Task.cpp:94\n       0xbfc6c0a\tDB::(anonymous namespace)::Spinner::awaitAndPushReadyTask(std::__1::unique_ptr<DB::Task, std::__1::default_delete<DB::Task> >&&) [tiflash+201092106]\n                \tdbms/src/Flash/Pipeline/Schedule/WaitReactor.cpp:43\n       0xbfc6470\tDB::WaitReactor::loop() [tiflash+201090160]\n                \tdbms/src/Flash/Pipeline/Schedule/WaitReactor.cpp:157\n       0xbfc7b81\tdecltype(*(static_cast<DB::WaitReactor*>(fp0)).*fp()) std::__1::__invoke<void (DB::WaitReactor::*)() noexcept, DB::WaitReactor*, void>(void (DB::WaitReactor::*&&)() noexcept, DB::WaitReactor*&&) [tiflash+201096065]\n                \t/usr/local/bin/../include/c++/v1/type_traits:3859\n       0xbfc7b2d\tvoid std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::WaitReactor::*)() noexcept, DB::WaitReactor*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::WaitReactor::*)() noexcept, DB::WaitReactor*>&, std::__1::__tuple_indices<2ul>) [tiflash+201095981]\n                \t/usr/local/bin/../include/c++/v1/thread:280\n       0xbfc789b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::WaitReactor::*)() noexcept, DB::WaitReactor*> >(void*) [tiflash+201095323]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f954c792802\tstart_thread [libc.so.6+653314]"] [source=BaseDaemon] [thread_id=906]
```
### 4. What is your TiFlash version? (Required)
master
```
[2023/04/27 01:28:38.143 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tikv] [version=7.2.0-alpha] [git_hash=8656623b8b9c9a590b9f61aedcce23ee38ed5023]
[2023/04/27 01:28:38.143 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tidb] [version=7.2.0-alpha] [git_hash=91bf358f7ef10517e7794af8c6146cc854a36b69]
[2023/04/27 01:28:38.143 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tiflash] [version=7.2.0-alpha] [git_hash=d313fa5071ea17b1c59bfae7a8be063a8bea87a7]
[2023/04/27 01:28:38.143 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=pd] [version=7.2.0-alpha] [git_hash=400e3bd30c563bdb6b18ee7d59e56d082afff6a6]
```

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #7395

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Operators/SharedAggregateRestorer.h [hunk 1]
```diff
@@ -96,18 +96,26 @@ class SharedSpilledBucketDataLoader : public std::enable_shared_from_this<Shared
 };
 using SharedSpilledBucketDataLoaderPtr = std::shared_ptr<SharedSpilledBucketDataLoader>;
 
+enum class SharedLoadResult
+{
+    success,
```
Comment (yibin87) [inline dbms/src/Operators/SharedAggregateRestorer.h:101]: why not use upper case names here?

**Review by yibin87 — APPROVED** [review]
others LGTM

File: dbms/src/Operators/SharedAggregateRestorer.h (hunk 1 above)
Comment (SeaRise) [reply dbms/src/Operators/SharedAggregateRestorer.h]: ok, updated.

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

**Pull request merged**